dkg-gadget/src/gossip_engine/network.rs

```diff
@@ -403,15 +414,8 @@ impl<B: Block + 'static> GossipHandler<B> {
 			match self.pending_messages_peers.entry(message.message_hash::<B>()) {
 				Entry::Vacant(entry) => {
 					log::debug!(target: "dkg_gadget::gossip_engine::network", "NEW DKG MESSAGE FROM {}", who);
-					let recv_count = self.controller_channel.receiver_count();
-					if recv_count == 0 {
-						log::warn!(target: "dkg_gadget::gossip_engine::network", "No one is going to process the message!!!");
-					}
-					if let Err(e) = self.controller_channel.send(message.clone()) {
-						log::error!(target: "dkg_gadget::gossip_engine::network", "Failed to send message to DKG controller: {:?}", e);
-					} else {
-						log::debug!(target: "dkg_gadget::gossip_engine::network", "Message sent to {recv_count} DKG controller listeners");
-					}
+					let mut queue_lock = self.message_queue.write();
```

Comment:
Consider adding a TODO to do some validation here over messages: https://github.com/webb-tools/dkg-substrate/issues/370